body.tex

```diff
@@ -0,0 +1,481 @@
+\item software test and verification: the environment includes, on top of the build and debug tools, additional tools to permit/facilitate the testing and validation activities.
+\item operations: the environment shall include only tools required for the execution of an operational product. It shall not contain any build or debug tools. It shall be optimized for the operational activities. \footnote{In case a problem is found in operation the first thing to do is to replicate it in a different environment, where also debug and further analysis is possible.}
+\end{itemize}
+
+These are the most common uses of an environment. Additional scenarios can be identified.
+
+In some cases, the same environment is used in development, tests, and operations.
+
+Operational environment definitions, including location-specific configuration, shall never be included in the software products.
+Vanilla environment definition can be provided for unit tests purposes, as an example or template.
+
+
+\subsection{Software Product} \label{sec:swprod}
+
+A SW product is a \textit{defined} and \textit{controlled} software implementation with a specific scope, inside an overall system.
+
+\begin{itemize}
+\item \textit{defined}: it shall be clear which is the scope of the software product
+\item \textit{controlled}: the source code shall be maintained in a software repository, like for example GitHub.
+\end{itemize}
+
+%A software product is a component of the subsystem (DM) product tree.
+
+The following activities are done on a SW Product:
+\begin{itemize}
+\item development activities
+\item test activities, usually unit tests and system test
+\item release activities
+\item packaging activities
+\item operational activities, in standalone mode, in an operational pipeline or as part of a service. 
+Some SW Products are libraries, therefore are not operationally but used as dependencies by other software products.
+\end{itemize}
+
+A software product shall correspond to a single repository.
+In other words, there shall be a correspondence \textit{1:1} between software products and Git repositories.
+
+This is not the case of DM, where a software product is comprised of multiple Git repositories.
+The compromised solution, in this case, is to have a Git \textit{meta-package} that identifies the software product.
+All Git repositories composing the software product are enumerated in a Git \textit{meta-package} and released at the same time.
+In this case, a Git package shall be related only to one software product, and it is not by himself a software product.
+In other words, there is a correspondence \textit{1:many} between the software product and Git repositories.
+There shall never be a \textit{many:many} correspondence.
+
+The following elements should not be part of a software product, but follow a parallel development/release process:
+
+\begin{itemize}
+\item build tools, otherwise we will have them deployed in operations.
```

Comment:
example?

Reply:
In this moment, we are tagging as part of the lsst_distrib software product, the packages `scons` and `sconsUtils`.

In operations (Data Processing) at NCSA, we shall not deploy those packages. Scons could probably be resolved from conda and included in the environment.

Of course, those packages need to be included in the distribution to the science community.